### ISSUE TITLE
SCEDC dataset is incomplete / corrupted in the remote repository?

### ISSUE THREAD

**Issue by liutianlin0121** [seisbench/seisbench#302]
First of all, thank you for this awesome repository! It is tremendously helpful for my research.

I noted some inconsistencies between the documentation of the SCEDC dataset and the actual dataset hosted [at the remote repository](https://dcache-demo.desy.de:2443/Helmholtz/HelmholtzAI/SeisBench/datasets/scedc). Assuming the documentation is correct, this could mean that the dataset stored at the remote is incomplete or not provided as intended. Here are the inconsistencies I observed:

## Observation 1: Data size difference
In the official documentation [here](https://seisbench.readthedocs.io/en/stable/pages/benchmark_datasets.html#scedc), the dataset size of SCEDC is:

>Dataset size: waveforms.hdf5 ~660Gb, metadata.csv ~2.2Gb

However, in the dataset stored [at the remote repository](https://dcache-demo.desy.de:2443/Helmholtz/HelmholtzAI/SeisBench/datasets/scedc), the sizes are smaller: ~637Gb for waveforms.hdf5 and ~2.0Gb for metadata.csv.

<img width="890" alt="Screenshot 2024-06-30 at 23 40 19" src="https://github.com/seisbench/seisbench/assets/10226549/36d9b177-5624-48d5-b944-2e3cd30403fe">

## Observation 2: Difference in the number of traces
Table 1 of the [Seisbench paper](https://arxiv.org/pdf/2111.00786) mentions 8,111,060 traces in SCEDC. However, there are only 8,035,833 traces in the actual SCEDC dataset. Below is the metadata printed via:

```python
import pandas as pd

df = pd.read_csv('.seisbench/datasets/scedc/metadata.csv')
df
```

<img width="688" alt="Screenshot 2024-07-01 at 04 56 39" src="https://github.com/seisbench/seisbench/assets/10226549/65583e58-8dfb-477a-8226-25529885afc0">

I also checked the metadata for other datasets. For them, the number of traces matches exactly with those reported in Table 1 of the paper. The inconsistency appears only for SCEDC.

<img width="676" alt="Screenshot 2024-07-01 at 05 03 35" src="https://github.com/seisbench/seisbench/assets/10226549/52f4d201-f05d-40fc-922f-3b4a699ba426">

## Observation 3: Some components of SCEDC datasets are zero
I plotted some waveforms of the SCEDC dataset using:

```python
import seisbench.data as sbd
import matplotlib.pyplot as plt

dataset = sbd.SCEDC(
      sampling_rate=100,
      component_order='ZNE',
      dimension_order='NCW',
      cache=None
)

for i in range(3):
  plt.figure(figsize=(3, 2))
  plt.plot(dataset.get_waveforms(i).T)
  plt.title(f'waveform index {i}')
```

The result shows that only one component is nonzero, while the other two are all zeros.

<img width="312" alt="Screenshot 2024-07-01 at 05 11 11" src="https://github.com/seisbench/seisbench/assets/10226549/13741d9a-430f-481f-9cc6-c7c27d13fe7a">

The results are strange, because for all these waveforms, only 1 component (the blue lines) is nonzero, and the other two are zeros. Are these zero-valued components expected for SCEDC? If not, could they indicate some errors in the dataset?

## Observation 4: Warnings when loading the SCEDC dataset

When I load the dataset with:

```python
dataset = sbd.SCEDC(
      sampling_rate=100,
      component_order='ZNE',
      dimension_order='NCW',
      cache=None
)
```

Two warnings appear:

```
2024-07-01 06:09:47,058 | seisbench | WARNING | No data_format group found in .hdf5 File.
2024-07-01 06:09:47,059 | seisbench | WARNING | Dimension order not specified in data set. Assuming CW.
```

These warnings only appear when I load the SCEDC dataset, not other datasets. Could they suggest the waveform.hdf5 is not properly pre-compiled or prepared?

Thank you for your help!

Tianlin

**Comment by yetinam** [comment]
Hi @liutianlin0121 ,

thanks for the detailed report. I think I can clear up most of the confusion. The good side is, that the dataset is not corrupted, the bad side that the details in the SeisBench paper seem to be slightly wrong. Regarding dataset size, this is because we transcoded the dataset to a different format at some point. My colleague writing the documentation however still used the original version and used its size. For the benchmark, we used the transcoded one. In this process, we also did an additional quality control that removed a few further traces. Traces with all-zero channels are one-component traces, i.e., the instrument did not record other channels. For the missing data format group, I'm surprised myself. Fortunately, the dataset follows the default settings here, so there will be no issues. However, we should eventually fix that and update the documentation.

I hope this answers your questions and once again, thanks for the detailed report.
